src/aleph/http/client.clj

```diff
@@ -396,7 +396,7 @@
 
                         :else
                         cause)]
-         (s/put! response-stream response)
+         (d/error! early-response-d response)
```

Comment:
Oh yeah, so far, I was unable to even reach this code path and have it result in an error response anyway. It would probably be a good idea to come up with a bunch of test cases for the proxy functionality (currently there are none).

Reply:
Given how untested proxy code is, I'd strongly suggest writing tests before approving.

Maybe whoever needed the proxies could lend a hand?

Reply:
I was able to come up with a test that fails as expected without the patch - sanity check welcome :pray: 

Using the test as a basis, I should be able to see how we can reach the error case referenced above. Stay tuned.

Reply:
OK turns out there was a bit of breakage in the proxy connection error handling and propagation due to the various recent changes in the client response pipeline. Pushed 9673ac980e8b2eca8be8f6377c22a59f10b89155 for fixing those (see commit message for details) and added two proxy connection error test cases which indeed do hit those code paths now :sweat_drops:

Reply:
Resolved as per you guys' approval of the changes :pray: